tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -397,11 +413,25 @@ def media_files(self):
 
     def delete_all_sms(self):
         self.marionette.switch_to_frame()
-        return self.marionette.execute_async_script("return GaiaDataLayer.deleteAllSms();", special_powers=True)
+        self.marionette.push_permission('sms', True)
+        with self.marionette.using_context('chrome'):
+            self.marionette.execute_script('Services.prefs.setBoolPref("dom.sms.enabled", true);')
+        result = self.marionette.execute_async_script("return GaiaDataLayer.deleteAllSms();")
+        self.marionette.push_permission('sms', False)
+        with self.marionette.using_context('chrome'):
+            self.marionette.execute_script('Services.prefs.clearUserPref("dom.sms.enabled");')
```

Comment:
As we have set/get pref methods, I wonder if it makes sense to have a `clear_user_prefs` method too.